tasks/src/main/scala/dagr/tasks/gatk/HaplotypeCaller.scala

```diff
@@ -47,25 +48,47 @@ class HaplotypeCaller(ref: PathToFasta,
  extends GatkTask("HaplotypeCaller", ref, intervals=intervals) {
 
   override protected def addWalkerArgs(buffer: ListBuffer[Any]): Unit = {
-    buffer.append("--minPruning", minPruning)
-    buffer.append("--maxNumHaplotypesInPopulation", "200")
-    buffer.append("-variant_index_parameter", "128000")
-    buffer.append("-variant_index_type", "LINEAR")
-    buffer.append("--emitRefConfidence", "GVCF")
-    ploidy.foreach(p => buffer.append("--sample_ploidy", p))
-    buffer.append("--max_alternate_alleles", maxAlternateAlleles)
-    maxReadsInRegionPerSample.foreach(n => buffer.append("--maxReadsInRegionPerSample", n))
-    minReadsPerAlignmentStart.foreach(n => buffer.append("--minReadsPerAlignmentStart", n))
-    buffer.append("--contamination_fraction_to_filter", contaminationFraction)
-    buffer.append("-I", bam.toAbsolutePath)
-    buffer.append("-o", vcf.toAbsolutePath)
 
-    if (rnaMode) {
-      buffer.append("-dontUseSoftClippedBases")
-      buffer.append("-stand_call_conf", 20)
-      buffer.append("-stand_emit_conf", 20)
-    }
+    gatkMajorVersion match {
+      case n if n <  4 =>
+        // Args for versions 1-3
+        buffer.append("-I", bam.toAbsolutePath)
+        buffer.append("-o", vcf.toAbsolutePath)
+        buffer.append("--minPruning", minPruning)
+        buffer.append("--maxNumHaplotypesInPopulation", maxHaplotypes)
+        buffer.append("--emitRefConfidence", "GVCF")
+        buffer.append("--max_alternate_alleles", maxAlternateAlleles)
+        buffer.append("--contamination_fraction_to_filter", contaminationFraction)
+        ploidy.foreach(p => buffer.append("--sample_ploidy", p))
+        if (useNativePairHmm) buffer.append("-pairHMM", "VECTOR_LOGLESS_CACHING")
+
+        buffer.append("-variant_index_parameter", "128000")
+        buffer.append("-variant_index_type", "LINEAR")
+        maxReadsInRegionPerSample.foreach(n => buffer.append("--maxReadsInRegionPerSample", n))
+        minReadsPerAlignmentStart.foreach(n => buffer.append("--minReadsPerAlignmentStart", n))
+
+        if (rnaMode) {
+          buffer.append("-dontUseSoftClippedBases")
+          buffer.append("-stand_call_conf", 20)
+          buffer.append("-stand_emit_conf", 20)
+        }
 
-    if (useNativePairHmm) buffer.append("-pairHMM", "VECTOR_LOGLESS_CACHING")
+      case n if n >= 4 =>
+        // Args for version 4
+        buffer.append("-I", bam.toAbsolutePath)
+        buffer.append("-O", vcf.toAbsolutePath)
```

Comment:
In `GenotypeGvcfs` you have a section before this block that adds common arguments:
```scala
// Args that are common to all versions
```
I'd recommend moving common args similarly, or not have common args in `GenotypeGvcfs`

Reply:
I almost did it, but in fact there's only _1_ common arg, and it felt not worth it.  Note that output is `-o` vs.`-O`.  But I can move the `-I` argument out.